e2e/index.ts

```diff
@@ -27,13 +27,20 @@ async function main() {
     throw new Error(`Test plan ${testPlaneName} not found`);
   }
 
+  const ensureDeletion = !!process.env.ENSURE_DELETION;
   const stack = await LocalWorkspace.createOrSelectStack({
     projectName: 'yoga-e2e',
     stackName: identifier,
     program: testPlan.program,
   });
 
   try {
+    // In case of deletion, we need to ensure that the stack is deleted only
+    // so we throw here and the "finally" block will take care of deletion, if possible.
+    if (ensureDeletion) {
+      throw new Error('Ensure deletion');
```

Comment:
Doesn't this always mark the CI as failed ?

Reply:
`finally` block will capture this and do deletion for the stack